src/GafferScene/PrimitiveVariableTweaks.cpp

```diff
@@ -0,0 +1,769 @@
+{
+	return
+		ObjectProcessor::affectsProcessedObject( input ) ||
+		input == interpolationPlug() ||
+		input == selectionModePlug() ||
+		input == idListPlug() ||
+		input == idListVariablePlug() ||
+		input == idPlug() ||
+		input == maskVariablePlug() ||
+		input == ignoreMissingPlug() ||
+		tweaksPlug()->isAncestorOf( input )
+	;
+}
+
+void PrimitiveVariableTweaks::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	if( tweaksPlug()->children().empty() )
+	{
+		h = inPlug()->objectPlug()->hash();
+	}
+	else
+	{
+		ObjectProcessor::hashProcessedObject( path, context, h );
+		interpolationPlug()->hash( h );
+		selectionModePlug()->hash( h );
+		idListPlug()->hash( h );
+		idListVariablePlug()->hash( h );
+		idPlug()->hash( h );
+		maskVariablePlug()->hash( h );
+		ignoreMissingPlug()->hash( h );
+		tweaksPlug()->hash( h );
+	}
+}
+
+IECore::ConstObjectPtr PrimitiveVariableTweaks::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, const IECore::Object *inputObject ) const
+{
+	const Primitive *inputPrimitive = runTimeCast<const Primitive>( inputObject );
+	if( !inputPrimitive || tweaksPlug()->children().empty() )
+	{
+		return inputObject;
+	}
+
+	PrimitiveVariable::Interpolation targetInterpolation = (PrimitiveVariable::Interpolation)interpolationPlug()->getValue();
+
+	const bool ignoreMissing = ignoreMissingPlug()->getValue();
+
+	PrimitivePtr result = inputPrimitive->copy();
+
+	SelectionMode selectionMode = (SelectionMode)selectionModePlug()->getValue();
+	ConstInt64VectorDataPtr idList;
+	if( ( selectionMode == SelectionMode::IdList || selectionMode == SelectionMode::IdListPrimitiveVariable ) && targetInterpolation != PrimitiveVariable::Invalid )
+	{
+		if( selectionMode == SelectionMode::IdList )
+		{
+			idList = idListPlug()->getValue();
+		}
+		else
+		{
+			std::string idListVarName = idListVariablePlug()->getValue();
+			auto idListVar = inputPrimitive->variables.find( idListVarName );
+			if( idListVar == inputPrimitive->variables.end() )
+			{
+				throw IECore::Exception( fmt::format( "Can't find id list primitive variable \"{}\".", idListVarName ) );
+
+			}
+
+			if( idListVar->second.interpolation == PrimitiveVariable::Interpolation::Constant )
+			{
+				if( const Int64VectorData *int64Data = IECore::runTimeCast<Int64VectorData>( idListVar->second.data.get() ) )
+				{
+					idList = int64Data;
+				}
+				else if( const IntVectorData *intData = IECore::runTimeCast<IntVectorData>( idListVar->second.data.get() ) )
+				{
+					// For simplicity elsewhere, just convert to an Int64VectorData instead of supporting both types
+					Int64VectorDataPtr convertedData = new Int64VectorData();
+					auto &converted = convertedData->writable();
+					converted.reserve( intData->readable().size() );
+					for( int i : intData->readable() )
+					{
+						converted.push_back( i );
+					}
```

Comment:
I think this can be a one-liner? 

```
idList = new Int64VectorData( vector<int64_t>( intData->readable().begin(), intData->readable().end() ) );
```